src/ocaml/typing/untypeast.ml

```diff
@@ -517,7 +517,9 @@ let expression sub exp =
                              ])
     | Texp_open (od, exp) ->
         Pexp_open (sub.open_declaration sub od, sub.expr sub exp)
-    | Texp_hole -> Pexp_hole
+    | Texp_hole ->
+        let id = Location.mkloc "merlin.hole" loc in
+        Pexp_extension (id, PStr [])
```

Comment:
We've imported `Ast_helpers`, so this can just be `Exp.hole ()`, I believe

Reply:
`Exp.hole ()` has type `expression` but `expression_desc` is expected here so we cannot use it. It is probably enough to only factor the `merlin.hole` name out.